### ISSUE TITLE
Implementation of internal branching/Folgezettel

### ISSUE THREAD

**Issue by pietroiusti** [org-roam/org-roam#1451]
First of all, thank you for your great work.

I write this to propose the implementation of "internal branching" or
"Folgezettel", a feature of Luhmann's slip-box. 

For those who don't know the idea here is a brief description. Suppose
we have note 1. If Luhman wanted to /continue/ the thought in note 1,
he would have added note 2, after note 1. If he wanted to continue the
thought in 2 he would have added note 3. And so on. But, if he wanted
to /supplement/ something present in note 1, he would have added note
1a, between note 1 and note 2. If we wanted to continue the thought in
note 1a, he would have added note 1b, a continuation of note 1a,
between note 1a and 2. And so
on. (cf. http://luhmann.surge.sh/communicating-with-slip-boxes where
he talks of "the possibility of arbitrary internal branching") We
would, then, have the slips in the following order:

1, 1a, 1b, 2, 3.

Give their relations, they can be represented like this:

\* 1
** 1a
** 1b
\* 2
\* 3
 (cf. how Luhmann's archive is representing his slip-box:
 https://niklas-luhmann-archiv.de/bestand/zettelkasten/inhaltsuebersicht).

Org-roam cannot represent the content of the slip-box in this way,
because it allows links, but not internal branching. (Well, to be
fair, we can declare the relationships within the slips, but org-roam
doesn't recognize them so to say.).

In the "zettlekasten" community there seem to be disagreements as to
whether this feature is actually something good (see the discussion
here, for example:
https://zettelkasten.de/posts/luhmann-folgezettel-truth/). Some argue
that it provides benefits over linking alone. Others think that it
doesn't add anything valuable.  Regardless of who is right, I want to
point out that this feature could be implemented, I think, rather
cheaply, and in a non-intruding way.  It would add something that many
consider valuable and detract nothing from those who consider it
useless or harmful.

We only need to introduce two special meanings that can be attributed to
links: "first-daughter-of" and "next-sister-of" (or, if you prefer,
"complementation-of" and "continuation-of"). This can be done, for
example with a line at the beginning of a note like

#+roam_first_daughter_of: linkToMotherNote

or by introducing some special notation, something like
"{{first_daughter_of}linkToMotherNote}". I don't know what would be
more convenient from a technical point of view.

Consider how this would allow creating the linear structure
represented in the example above. You create a note "foo" (note
1). You want to create another note that /continues/ the thought in
"foo", so you create note "foobar" (note 2), so you insert in "foobar"
a link to "foo" with the special meaning of next-sister-of. Now you
want to create a note that continues "foobar". So you create a note
"foobarbaz" (note 3) and you put in it a link to "foobar" with the
special meaning of next-sister-of. Now you want to /add/ something to
note "foo". You then create a note "daugtherfoo" (note 1a) with a link
to "foo" with the special meaing of "first-daughter-of". Finally,
ending the example, you create a note "daughterfoobar" (note 1b) that
continues "daughterfoo", by inserting in it a link to "daughterfoo" with 
the special meaning of next-sister-of.

Of course, a note can have only one first-daughter-of link and only
one next-sister-of link. (Also, placing a sister between two sisters should
be handled one way or another.)

The presence of these special links would allow for an alternative way
of representing the entire slip-box (and hence an alternative way to
navigate it), based on the "relationships between relatives":

\* foo
** daugtherfoo
** daugtherfoobar
\* foobar
\* foobarbaz

This can be represented in a normal emacs buffer and it can be a
useful way to navigate the slip-box and find a place for new
notes. Those notes that have no special links would be represented as
one-star nodes (in any order). In the graph org-roam is already able
to create, special links would count as normal links, as in any other
context.

To summarize:
- we would have a feature that some people deem important, is
  implemented in other pieces of software, and was a feature of
  Luhmann's original method.
- no need to change any existing feature;
- it would not get in the way of those who don't want to use it;
- rather easy to implement (I suppose! :)), we only need to introduce
  two special meanings for links.
- it would allow for a further way to display and navigate the
  slip-box.

**Comment by jethrokuan** [comment]
Rather than implementing a special link for this, what do you think of just using the outline hierarchy as you gave above?

```
We
would, then, have the slips in the following order:

1, 1a, 1b, 2, 3.

Give their relations, they can be represented like this:

* 1
** 1a
** 1b
* 2
* 3
```

In v2, headlines become the most granular level, and we have the luxury of using Org's outline hierarchy compared to other non-org solutions.

**Comment by pietroiusti** [comment]
@jethrokuan 
Fantastic! Directly using Org's built-in outline hierarchy seems a much
better idea than using special links.

We would be able to insert/move notes and create the linear structures
we want in an easy, and familiar, way.

We would not have to implement anything about sisterhood
relationships: we would just place notes next to each other in the
order we like; and insert links just in case we want the connections
between sisters to appear in the graph, or if we want them to be
explicit inside the notes.

Daughterhood relationships (org children nodes), I think, should be
displayed in the graph as normal links between daughter and mother
nodes without the need to manually insert a link (and if a link is
inserted nonetheless, nothing would change in the graph).

(I think this gets very very close to the original slip-box, but with a
flexibility that was not possible in Luhmann's physical implementation.)

As far as I understand, we already have some of these functionalities
in v2.

What do you think?
